core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -489,8 +493,9 @@ class ReplicaManager(val config: KafkaConfig,
       val fetchPartitionStatus = logReadResults.map { case (topicAndPartition, result) =>
         (topicAndPartition, FetchPartitionStatus(result.info.fetchOffsetMetadata, fetchInfo.get(topicAndPartition).get))
       }
+
       val fetchMetadata = FetchMetadata(fetchMinBytes, fetchOnlyFromLeader, fetchOnlyCommitted, isFromFollower, fetchPartitionStatus)
-      val delayedFetch = new DelayedFetch(timeout, fetchMetadata, this, responseCallback)
+      val delayedFetch = new DelayedFetch(timeout, fetchMetadata, this, quota ,responseCallback)
```

Comment:
Yes, the follower has a similar issue. There is already logic to add a delay per partition. So, if a partition is throttled in the follower, we can calculate a delay from quota and delay the partition accordingly. We probably also need to change the backoff logic in AbstractFetcherThread a bit. Instead of always backing off for a fixed amount of time, it's probably better to backoff based on the smallest delay among all partitions.

I agree that this is probably not a common issue. It only becomes a big issue if the maxWait or replica backoff time are configured very large (say close to the metric window * sample size). So, we can address that in a followup jira.